src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/LiveShareProjectSnapshotBase.cs

```diff
@@ -27,6 +27,10 @@ internal class LiveShareProjectSnapshotBase : ProjectSnapshot
 
         public override DocumentSnapshot GetDocument(string filePath) => throw new NotImplementedException();
 
+        public override bool IsImportDocument(DocumentSnapshot document) => throw new NotImplementedException();
```

Comment:
This is breaking to LiveShare, we need to mark these as virtual or work with the LiveShare team to coordinate a release.

Reply:
When this that code move to this repo? Also isn't live share currently broken?

Reply:
> When this that code move to this repo? 
When they tell us we can open source it T_T

> Also isn't live share currently broken?

Nah, it was fixed

Reply:
`LiveShareProjectSnapshotBase` is in this repo - I'm going to mark it virtual there.

We should try to eventually get rid of all of the subclasses of PSM - it's not meant to be an abstraction.

Reply:
I should clarify, the that if there are features of PSM that we need to have different implementations, then **those** should have services that differ across hosts, rather than subclassing.

Reply:
`LiveShareProjectSnapshotBase` already overrides these - does that resolve your concern?